src/server/kernel/wseat.cpp

```diff
@@ -621,6 +726,105 @@ void WSeat::notifyFrame(WCursor *cursor)
     d->doNotifyFrame();
 }
 
+// top level touch event
+
+void WSeat::notifyTouchDown(WCursor *cursor, WInputDevice *device, int32_t touch_id, uint32_t time_msec)
+{
+    W_D(WSeat);
+    auto qwDevice = qobject_cast<QPointingDevice*>(device->qtDevice());
+    Q_ASSERT(qwDevice);
+    const QPointF &globalPos = cursor->position();
+
+    qDebug() << "TouchDown: tid" <<  touch_id << " pos:" << globalPos << " time:" << time_msec;
+    auto *state = &d->devState[qwDevice];
+    QWindowSystemInterface::TouchPoint *tp = state->point(touch_id);
+    if (tp) {
+        qWarning("Incorrect touch state");
+    } else {
+        QWindowSystemInterface::TouchPoint newTp;
+        newTp.id = touch_id;
+        newTp.state = QEventPoint::Pressed;
+        newTp.area = QRect(0, 0, 8, 8);
```

Comment:
Why is `0, 0, 8, 8`

Reply:
same with qtwayland
https://codebrowser.dev/qt6/qtbase/src/platformsupport/input/libinput/qlibinputtouch.cpp.html#113

Reply:
Add some annotations here